### ISSUE TITLE
[20] Distilling Self-Supervised Vision Transformers for Weakly-Supervised Few-Shot Classification & Segmentation

### ISSUE THREAD

**Issue by sy00n** [sy00n/DL_paper_review#22]
## Abstract
- 본 논문은 self-supervised pretrained ViT를 활용한 weakly-supervsed few-shot image classification & segmentation task를 다룬다. 
- pixel-level label 없이 오직 image-level label만 가지고 classfication과 segmentation을 수행한다. 
- 이를 위해 pre trained ViT backbone에서의 token에 해당하는 attention map을 사용해서 pseudo-label로 사용한다. 
- 더 실용적인 setup인 "mixed" supervision을 사용해서 groud truth가 존재하는 작은 수의 training 이미지와 나머지 only image-level label만 가지는 이미지 둘 다 사용한다. 
- Mixed setup에서는 pesudo-label enhancer를 사용해서 pseudo-label을 개선한다. 

![image](https://github.com/sy00n/DL_paper_review/assets/67910856/874b7cb2-70be-436c-b32e-73ae78d4f76d)

## Method
![image](https://github.com/sy00n/DL_paper_review/assets/67910856/5322cbae-aad5-4248-b147-2a53a976c932)
N개의 class에 대한 k개의 labeled support image가 있다.  FS-CS(few-Shot Classification and Segmentation) 모델은 1) query image를 각 class 별로 identify 할 수 있어야 하고 2) associated pixel을 segment 할 수 있어야 한다. 보통 이러한 셋팅을 N-way K-shot episode라고 불린다. 본 논문은 meta-learning approach를 적용해서 annotated training dataset을 통해 N-way k-shot episode sequence를 생성했다.    
- 본 논문에서의 셋팅 용어 정리
    - pixel-level supervision : ground-truth segmentation masks가 모든 support/query image에 대해 학습 시에 주어지는 경우
    - image-level supervision : class label만 접근 가능한 경우
    - mixed supervision : traininig image 중에 오작 작은 퍼센트만 ground-truth segmentation mask가 주어지는 경우 

### Classification-Segmentation transformer
- 본 논문에서의 Classification-Segmentation Transformer(CST)는 supprot image, query image 쌍 사이의 DINO ViT token의 cross correlation을 계산한다. 그 다음에 two new set token을 생성하고 이는 image classification / segmentation head 의 input으로 들어간다. 그리고 각 task head로부터 loss를 합친다. 

### Correlation tokens
쿼리 셋, 서포트 셋 마다 각각 patch token과 한 개의 class token이 있다. {쿼리 셋에 대한 패치 토큰, 이미지 토큰}, {서포트 셋에 대한 패치 토큰, 이미지 토큰}을 각각 {F_q, h_q}, {F_s, h_s}라고 정의한다.  쿼리 이미지 토큰 F_q에 대해 h_s(랑 F_s 각각 cosine similarity 계산을 통해 아래와 같이 cross correlation을 구한다. 
![image](https://github.com/sy00n/DL_paper_review/assets/67910856/32af8fe1-1627-4f80-b1f7-8d178c7f0f4a)
F와 는 각각 l2-normalized token이다. 그리고 ViT backbone에 M 개의 mutli-head가 있으면 이 M개의 head에 대해 각각 따로 correlation token을 구한다. (이전 연구에서 multi-layer semantics를 통합하는 것의 중요성을 보여준 바 있음)
유사하게, 본 논문에서도 L개의 trancformer layer에 대해 각각 correlation token을 구한다. ML개의 differenct set의 correlation token을 구하고 new dimension을 따라 concet 한다. 
![image](https://github.com/sy00n/DL_paper_review/assets/67910856/b454c5ef-9da0-4c2f-83ba-29eb70108898)

### Correlation transformer
c'_cls와 C'_img를 다 concat해서 Z를 만든다. correlation transformer는 2-layer transformer이어서 입력으로 Z와 supprot image에 대한 segmentation mask label을 받고 2개의 토큰(classification task token, segmentation task token)을 반환한다. 
![image](https://github.com/sy00n/DL_paper_review/assets/67910856/c4618acf-224a-43a6-bd3b-9f27a70bfbe7)
MSA는 multi-head self-attention, Norm은 group normalization , MLP는 한 개의 linear layer를 의미한다. j는 transformer layer index를 의미한다. 추가적으로
